frontend/src/store/VideoPlayer/index.js

```diff
@@ -207,6 +209,30 @@ class VideoPlayer {
   toggleFullscreen = () => {
     this.isFullscreen = !this.isFullscreen
   }
+
+  @action
+  onShowModal = (callback) => {
+    this.showModal = true
+    if(typeof callback === 'function') {
+      this.showModalCallback = callback
+    }
+  }
+
+  @action
+  onStopStreamingConfirm = async () => {
+    await this.videoEl.current.pause()
+    this.showModal = false
```

Comment:
Seems like both `onStopStreamingConfirm` and `onStopStreamingCancel` do the same cleanup stuff like resetting the modal state. Do we want just to extract this logic into separate action?

Reply:
both methods use lines `this.showModal = false` and `this.showModalCallback = null`, but onStopStreamingConfirm does that invoking `showModalCallback` between these 2 lines, so we can't use just one method for resetting modal state